### ISSUE TITLE
Undiagnosed Template Haskell error (T11721_TH)

### ISSUE THREAD

**Issue by aspiwack** [tweag/ghc#236]
Here is one of the errors. It's rather obscure. I don't know what's happening.

```
T11721_TH.hs:15:5: error:
    • T11721_TH failed
    • In the untyped splice:
        $(do let rightOrder :: [TyVarBndr] -> Bool
                 rightOrder [KindedTV b _, KindedTV a _]
                   = nameBase b == "b" && nameBase a == "a"
                 rightOrder _ = False
             TyConI (DataD _ _ _ _ [ForallC con_tvbs1 _ _] _) <- reify ''T
             DataConI _ (ForallT con_tvbs2 _ _) _ <- reify 'MkT
             if rightOrder con_tvbs1 && rightOrder con_tvbs2 then
                 [| () |]
             else
                 fail "T11721_TH failed")

*** unexpected failure for T11721_TH(normal)
```

**Comment by monoidal** [comment]
The test can be made to pass in `fix-th-test-2` (07dfa0f7ded), but this is backwards incompatible.

**Comment by mboes** [comment]
@monoidal can you confirm that this is now fixed?
